src/main/scala/files_metadata/infrastructure/FilesMetaPostgresRepository.scala

```diff
@@ -589,5 +589,66 @@ class FilesMetaPostgresRepository extends FilesMetaRepository {
     }
   }
 
-  override def deleteFileMeta( ownerUuid: UUID, uuid: UUID ): Unit = ???
+  override def deleteFileMeta( uuid: UUID ): Unit = {
+    val connection: Connection = pool.getConnection()
+    connection.setAutoCommit( false )
+
+    try {
+      val getArchiveStatement = connection.prepareStatement(
+        "SELECT archive_uuid FROM files WHERE uuid = ?"
+      )
+      getArchiveStatement.setObject( 1, uuid )
+      val resultSet = getArchiveStatement.executeQuery()
+      if (resultSet.next()) {
+        val archive_uuid =
+          resultSet.getObject( "archive_uuid" ).asInstanceOf[UUID]
+        val deleteArchiveFilesStatement = connection.prepareStatement(
+          "DELETE FROM archives WHERE uuid = ?"
+        )
+        deleteArchiveFilesStatement.setObject( 1, archive_uuid )
+        deleteArchiveFilesStatement.executeUpdate()
+      }
+      connection.commit()
+    } finally {
+      connection.close()
+    }
+  }
+
+  override def deleteDirectoryMeta( uuid: UUID ): Unit = {
+    val connection: Connection = pool.getConnection()
+    connection.setAutoCommit( false )
+
+    try {
+      val statement = connection.prepareStatement(
+        """
+                SELECT uuid , archive_uuid
+                FROM files_view WHERE
+                parent_uuid = ?
+                """.stripMargin
+      )
+      statement.setObject( 1, uuid )
+
+      val result = statement.executeQuery()
+
+      // Delete each file one by one
+      while (result.next()) {
+        val fileUuid    = UUID.fromString( result.getString( "uuid" ) )
+        val archiveUuid = result.getString( "archive_uuid" )
+        if (archiveUuid == null) {
+          deleteDirectoryMeta( fileUuid )
+        } else {
+          deleteFileMeta( fileUuid )
+        }
+      }
+      println( uuid )
```

Comment:
Delete the `println`

Reply:
I'm sorry I did'n see it. Has been eliminated